File: java/mobil.java
public class Mobil {
private String merk;
private String mesin;
private String aksi;
private String rem;

```
// Constructor
public Mobil(String merk) {
    this.merk = merk;
    this.mesin = null;
    this.aksi = null;
    this.rem = null;
}

// Setter mesin
public void setMesin(String mesin) {
    this.mesin = mesin;
}

// Setter aksi
public void setAksi(String aksi) {
    this.aksi = aksi;
}

// Setter rem
public void setRem(String rem) {
    this.rem = rem;
}

// Getter merk
public String getMerk() {
    return merk;
}

// Method jalan
public void jalan() {
    if (mesin == null || aksi == null) {
        throw new IllegalArgumentException(
            "Mesin dan aksi tidak boleh null saat mobil berjalan."
        );
    }

    System.out.println("Mobil " + merk + 
                       " dengan mesin " + mesin + 
                       " sedang " + aksi + ".");
}

// Method pengereman
public void pengereman() {
    if (rem == null) {
        throw new IllegalArgumentException(
            "Informasi rem tidak boleh null saat pengereman."
        );
    }

    System.out.println("Mobil " + merk + 
                       " melakukan pengereman menggunakan " + rem + ".");
}
```

}

File: php/Mobil.php
<?php

class Mobil {

    // Properties / Field / Atribut
    public $merk;
    public $mesin;
    public $aksi;
    public $rem;

    // Constructor
    public function __construct($merk) {
        $this->merk = $merk;
    }

    // Method jalan
    public function jalan() {

        // Invariant: mesin dan aksi tidak boleh null
        if ($this->mesin === null || $this->aksi === null) {
            throw new InvalidArgumentException(
                "Mesin dan aksi tidak boleh null saat method jalan() dipanggil."
            );
        }

        echo "\n------>> Mobil Sedang Berjalan <<------\n";
        echo "Merk  : " . $this->merk . "\n";
        echo "Mesin : " . $this->mesin . "\n";
        echo "Aksi  : " . $this->aksi . "\n";
    }

    // Method pengereman
    public function pengereman() {

        // Invariant: rem tidak boleh null
        if ($this->rem === null) {
            throw new InvalidArgumentException(
                "Rem tidak boleh null saat method pengereman() dipanggil."
            );
        }

        echo "\n------>> Mobil Melakukan Pengereman <<------\n";
        echo "Merk : " . $this->merk . "\n";
        echo "Rem  : " . $this->rem . "\n";
    }
}

?>
